src/itest/kotlin/nl/lifely/zac/itest/ZaakRestServiceCompleteTest.kt

```diff
@@ -183,4 +189,104 @@ class ZaakRestServiceCompleteTest : BehaviorSpec({
             }
         }
     }
+
+    Given("A zaak has been created that has not finished the intake phase") {
+        lateinit var zaakUUID: UUID
+        lateinit var resultaatTypeUuid: UUID
+        val afhandelenId: Int
+
+        zacClient.createZaak(
+            zaakTypeUUID = ZAAKTYPE_INDIENEN_AANSPRAKELIJKSTELLING_DOOR_DERDEN_BEHANDELEN_UUID,
+            groupId = TEST_GROUP_A_ID,
+            groupName = TEST_GROUP_A_DESCRIPTION,
+            startDate = DATE_TIME_2000_01_01
+        ).run {
+            val responseBody = body!!.string()
+            logger.info { "Response: $responseBody" }
+            JSONObject(responseBody).run {
+                getJSONObject("zaakdata").run {
+                    zaakUUID = getString("zaakUUID").run(UUID::fromString)
+                }
+            }
+        }
+        itestHttpClient.performGetRequest(
+            "$ZAC_API_URI/zaken/resultaattypes/$ZAAKTYPE_INDIENEN_AANSPRAKELIJKSTELLING_DOOR_DERDEN_BEHANDELEN_UUID"
+        ).run {
+            val responseBody = body!!.string()
+            logger.info { "Response: $responseBody" }
+            JSONArray(responseBody).getJSONObject(0).run {
+                // we do not care about the specific result type, so we just take the first one
+                resultaatTypeUuid = getString("id").let(UUID::fromString)
+            }
+        }
+        itestHttpClient.performGetRequest(
+            "$ZAC_API_URI/planitems/zaak/$zaakUUID/userEventListenerPlanItems"
+        ).run {
+            val responseBody = body!!.string()
+            logger.info { "Response: $responseBody" }
+            JSONArray(responseBody).getJSONObject(0).run {
+                afhandelenId = getString("id").toInt()
+            }
+        }
+
+        When("the zaak is completed as 'not-admissible'") {
+            val receiverMail = "receiverZaakComplete@example.com"
+            val senderMail = "sender@example.com"
+            val mailBody = "<p><b>bold</b>paragraph<i>italic</i></p>"
+
+            // Wait before setting the status of a zaak (implicitly)
+            // because OpenZaak does not allow setting multiple statuses for one zaak
+            // within the same timeframe of one second.
+            // If we do not wait in these cases we get a 400 response from OpenZaak with:
+            // "rest_framework.exceptions.ValidationError: {'non_field_errors':
+            // [ErrorDetail(string='De velden zaak, datum_status_gezet moeten een unieke set zijn.', code='unique')]}"
+            //
+            // Related OpenZaak issue: https://github.com/open-zaak/open-zaak/issues/1639
+            sleep(1)
```

Comment:
Ah, unfortunate that we have to add another useless second to our integration test time.. I don't see a way to solve this unfortunately..

Reply:
We can do more open/close stuff, but then timewise it would be close to a second, while the tests won't be well separated.
So I decided 1 second is the better option :)